x-pack/dockerlogbeat/tools/setmount.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
+
+"""Manually build a docker plugin via a custom script"""
+
+import requests
+import sys
+import tempfile
+import os
+import io
+import tarfile
+import json
+import re
+import subprocess
+import shutil
+import argparse
+
+
+PLUGIN_URI = 'elastic/elastic-logging-plugin'
+API_URL = 'https://registry-1.docker.io/v2'
+
+USAGE = """
+Example Usage:
+
+./setmount.py /path/to/certs/on/docker/host
+
+You can also specify a custom name and version tag for the rebuild plugin via `--tag` and `--name`. By default, the rebuilt plugin will iterate the patch release of the latest plugin.
+By defualt, this script will build a new plugin based on the latest release available from the docker store. To specify a different version, use the `--from_release` argument.
+
+This script rebuilds a plugin and adds a custom bindmount path. This can be used to add user-supplied certificates to a Plugin via a mounted directory. 
+The host directory will be mounted as `/usr/local/share/ca-certificates` inside the plugin.
+
+After the bindmount has been created, you can adjust the bindmount source and destination via `docker plugin set`:
+
+$ docker plugin set elastic/elastic-logging-plugin:8.0.0 CUSTOM_DIR.source="/new/source/bindmount"
+
+In cases where you're building the plugin on a system that doesn't have access to the docker daemon where you want to run the plugin, 
+you can either use `docker plugin push` to push to your own docker repo (note that you must change the name with `--name`), or use the `--to_tar` flag with a target directory to generate a tarball.
+For example: `./setmount.py --to_tar /path/to/write/tar/new_plugin.tar.gz /path/to/certs/on/docker/host`
+To generate a docker plugin from the tarball, unpack the `.tar.gz` archive, and then run `docker plugin create NEW_PLUGIN_NAME PATH_TO_UNPACKED_DIRECTORY`.
+"""
+
+
+def cleanup(tmpfs: str):
+    """cleanup our tmpdir"""
+    print("cleaning up", tmpfs)
+    shutil.rmtree(tmpfs)
+
+
+def build_archive(tmpfs: str, target: str, name: str, tag: str):
+    """build a tar.gz archive of the plugin build manifest"""
+    print("Opening tar file.")
+    tar = tarfile.open(target, "w:gz")
+    print("Creating tar archive.")
+    tar.add(tmpfs + "/config.json", arcname="archive/config.json")
+    tar.add(tmpfs + "/rootfs", arcname="archive/rootfs")
+    tar.close()
+    print("====================================")
+    print("A tar archive has been created at ", target)
+    print("To build the plugin from the archive, unpack the tar.gz archive and run")
+    print("docker plugin create {}:{} archive/".format(name, tag))
+    print("====================================")
+    cleanup(tmpfs)
+
+
+def rebuild_plugin(tmpfs: str, name: str, tag: str):
+    """Rebuild the plugin via docker"""
+
+    # so we don't collide we anything else, iterate the semver string
+    reg = re.compile(r'''(\d+)\.(\d+)\.(\d+)''')
+    semver = list(reg.findall(tag))[0]
+    if len(semver) is not 3:
```

Comment:
`is` compares objects, better to use `!=` here (Python 3.8+ will output a warning)